DEVELOPMENT.md

```diff
@@ -2,88 +2,281 @@
+
+#### Testing
+
+Each component has its own test suite:
+
+```bash
+# Pages Tests
+cd pages
+npm test
+npm run test:e2e
+
+# Worker Tests
+cd worker
+npm test
+
+# Extension Tests
+cd extension
+npm test
+```
+
+For staging environment tests:
+```bash
+cd extension
+npm run test:e2e:staging
+```
+
 ## Architecture Overview
 
-### Frontend Stack
-- ES modules for code organization
-- IndexedDB for local storage
-- Environment-aware API endpoints
-- CSP and SRI for security
+### Component Architecture
+
+1. **Chrome Extension**
+   - Minimal popup window with launch button
+   - Main application in separate window for password manager support
+   - Background script for extension lifecycle management
+   - Content scripts for page interaction
+
+2. **Pages Application**
+   - React-based frontend
+   - Server-side rendering for better SEO
+   - API endpoints for data synchronization
+   - Admin interface for system management
+
+3. **Worker Service**
+   - Background sync service
+   - IndexedDB management
+   - Data processing and validation
+   - Health monitoring
+
+### Technology Stack
+
+1. **Extension**
+   - Manifest V3
+   - TypeScript
+   - React for UI
+   - Chrome Extension APIs
+
+2. **Pages**
+   - Next.js for SSR
+   - TypeScript
+   - Tailwind CSS
+   - API Routes
+
+3. **Worker**
+   - Node.js
+   - TypeScript
+   - IndexedDB
+   - WebSocket for real-time sync
 
 ### Security Implementation
-1. Content Security Policy
+
+1. **Extension Security**
+   - Manifest V3 security features
+   - Minimal popup with separate main window
+   - Password manager support
+   - Content script isolation
    ```javascript
-   const cspDirectives = [
-     "default-src 'self'",
-     "script-src 'self' 'unsafe-inline'",
-     "connect-src 'self' ${finalOrigin}"
-   ];
+   // manifest.json
+   {
+     "content_security_policy": {
+       "extension_pages": "script-src 'self'; object-src 'self'",
+       "sandbox": "sandbox allow-scripts allow-forms allow-popups"
+     }
+   }
    ```
 
-2. Module System
-   - ES modules with SRI validation
-   - Build-time integrity checks
-   - Explicit global assignments for event handlers
+2. **Pages Security**
+   - Content Security Policy (CSP)
+   - Subresource Integrity (SRI)
+   - CORS configuration
+   ```javascript
+   // next.config.js
+   {
+     async headers() {
+       return [{
+         source: '/:path*',
+         headers: [
+           {
+             key: 'Content-Security-Policy',
+             value: "default-src 'self'; script-src 'self'"
+           }
+         ]
+       }]
+     }
+   }
+   ```
 
-3. CORS Configuration
+3. **Worker Security**
+   - Origin validation
+   - Request authentication
+   - Data encryption
    ```javascript
-   const allowedDomains = [
-     'chroniclesync.xyz',
-     'chroniclesync-pages.pages.dev'
+   const allowedOrigins = [
+     'https://chroniclesync.xyz',
+     'https://staging.chroniclesync.xyz'
```

Comment:
staging what's that? We never deploy a staging like that. it's all feature branches